### PR TITLE
docs(discover): API docs for discover endpoint

### PR DESCRIPTION
**Pull request by shruthilayaj**
Screenshot 2024-07-31 at 10 02 14 AM
Screenshot 2024-07-31 at 10 02 05 AM
Screenshot 2024-07-31 at 10 00 28 AM
Screenshot 2024-07-31 at 10 00 19 AM
Screenshot 2024-07-31 at 10 00 11 AM

### CODE REVIEW

**Review by mjq — APPROVED** [review]

File: src/sentry/apidocs/parameters.py [hunk 1]
```diff
@@ -611,3 +611,40 @@ class DashboardParams:
...
+        type=str,
+        description="""The name of the Discover query you'd like to filter by.""",
+    )
+
+    SORT = OpenApiParameter(
+        name="sortBy",
+        location="query",
+        required=False,
+        type=str,
+        description="""The property to sort results by. If not specified the results are sorted by query name.
```
Comment (shruthilayaj) [inline src/sentry/apidocs/parameters.py:640]:
```suggestion
        description="""The property to sort results by. If not specified, the results are sorted by query name.
```

**Review by iamrajjoshi — APPROVED** [review]
lgtm, thanks for this 🙌🏽 , will defer to docs team for any wording suggestions

File: src/sentry/discover/endpoints/serializers.py [hunk 2]
```diff
@@ -143,18 +144,56 @@ def get_condition(self, condition):
...
+    range = serializers.CharField(
+        required=False,
+        allow_null=True,
+        help_text="The saved time range period for this saved query.",
+    )
+    fields = ListField(
+        child=serializers.CharField(),
+        required=False,
+        allow_null=True,
+        help_text="""The fields, functions, or equations to request for the query. At most 20 fields can be selected per request. Each field can be one of the following types:
```
Comment (lizokm) [inline src/sentry/discover/endpoints/serializers.py:180]:
```suggestion
        help_text="""The fields, functions, or equations that can be requested for the query. At most 20 fields can be selected per request. Each field can be one of the following types:
```

File: src/sentry/discover/endpoints/serializers.py [hunk 3]
```diff
@@ -143,18 +144,56 @@ def get_condition(self, condition):
...
+    - when a function is included, Discover will group by any tags or fields
+    - example: `field=count_if(transaction.duration,greater,300)`
+- An equation when prefixed with `equation|`. Read more about [equations here](/product/discover-queries/query-builder/query-equations/).
+    - example: `field=equation|count_if(transaction.duration,greater,300) / count() * 100`
+""",
+    )  # type: ignore[assignment]  # XXX: clobbers Serializer.fields
+    orderby = serializers.CharField(
+        required=False,
+        allow_null=True,
+        help_text="What to order the results of the query by. Must be something in the `field` list, excluding equations.",
```
Comment (lizokm) [inline src/sentry/discover/endpoints/serializers.py:195]:
```suggestion
        help_text="How to order the query results. Must be something in the `field` list, excluding equations.",
```

File: src/sentry/discover/endpoints/serializers.py [hunk 4]
```diff
@@ -168,13 +207,46 @@ class DiscoverSavedQuerySerializer(serializers.Serializer):
...
+- dailytop5
+- bar
+""",
+    )
+    topEvents = serializers.IntegerField(
+        min_value=1,
+        max_value=10,
+        required=False,
+        allow_null=True,
+        help_text="Number of top events timeseries to be visualized.",
```
Comment (lizokm) [inline src/sentry/discover/endpoints/serializers.py:245]:
```suggestion
        help_text="Number of top events' timeseries to be visualized.",
```

**Review by lizokm — APPROVED** [review]
Just a few tiny tweaks, otherwise looks good on my end :)

**Pull request merged**